src/components/quickLaunch/QuickLaunch.js

```diff
@@ -29,21 +31,30 @@ function QuickLaunch(props) {
     const icon = isPublic ? <PublicIcon /> : <LockIcon />;
     return (
         <Chip
+            clickable={clickable}
+            key={id}
             label={label}
             icon={icon}
             color={color}
             onDelete={handleDelete}
             onClick={handleClick}
             className={classes.chip}
             variant="outlined"
+            {...props}
```

Comment:
If you spread all the `props` down to the `Chip` component, then it will pass on any props it can't use to the underlying HTML component. Then React with throw a `Warning` that the underlying DOM element got props that it doesn't recognize (such as `isPublic` or `handleDelete`).

To avoid those warnings, you can destructure the function parameter, pulling out the props that `QuickLaunch` will use, and naming the remaining props something like `custom`. Then only those remaining `custom` props can be spread down to the `Chip` component:

```javascript
 function QuickLaunch(
    {
        id,
        label,
        clickable,
        handleClick,
        handleDelete,
        isDefault,
        isPublic,
        classes,
        ...custom,
    }
) {
    const color = isDefault || isPublic ? "primary" : "default";
    const icon = isPublic ? <PublicIcon /> : <LockIcon />;
    return (
        <Chip
            key={id}
            icon={icon}
            color={color}
            onDelete={handleDelete}
            onClick={handleClick}
            className={classes.chip}
            variant="outlined"
            {...custom}
         />
    );
}
```
Note that `label` and `clickable` will be spread from `custom`, so there's no need to pull those out in the function parameter destructuring.

Reply:
Thanks. I was about to go on a web hunt on how to fix those warning.

Reply:
I need to do something similar with advanced search, because I'm getting the same kind of prop warnings. Thanks for writing up this suggestion!